### PR TITLE
WIP: Transmission of data type in Tracking

### PR DESCRIPTION
**Pull request by zhaomr13**
Hi @mirguest ,

Could you please review the code and  test the building?
If it is ok, I will test the performance of this part with the help of Chengdong @fucd 

Mingrui.

### CODE REVIEW

**Comment by mirguest** [comment]
Hi Mingrui, 
Do you have any special settings? I found there is no build for your PR.
Tao

**Comment by zhaomr13** [comment]
Hi Tao @mirguest ,

I have no special setting. What do you mean "no build for your PR"? and what is PR?

Mingrui.

**Comment by mirguest** [comment]
Hi Mingrui,
PR means Pull Request. We have setup Travis-CI to build all the PRs automatically. But it looks like that the CI does not trigger in your Pull Request.
Tao

**Comment by zhaomr13** [comment]
How could the CI be triggered? Could you help to check?

**Comment by mirguest** [comment]
Hi Mingrui,
I have compiled your code and it looks fine. So I will merge your PR.
Tao

File: Service/TrackSystemSvc/TrackSystemSvc/MarlinTrkDiagnostics.h.remove [hunk 1]
```diff
@@ -17,12 +17,12 @@ namespace MarlinTrk{
   // LCIO Extension creating a pointer to the simhit for trackerhits 
   struct MCTruth4HitExtStruct{
     MCTruth4HitExtStruct() : simhit(0) {}
-    EVENT::SimTrackerHit* simhit;
```
Comment (mirguest) [inline Service/TrackSystemSvc/TrackSystemSvc/MarlinTrkDiagnostics.h.remove:20]: @zhaomr13 This file should be removed.

File: Service/TrackSystemSvc/src/DiagnosticsController.cc.remove [hunk 2]
```diff
@@ -419,9 +419,9 @@ namespace MarlinTrk{
     
     if(_skip_track) return;
     
-    EVENT::TrackerHit* trkhit = hit->getLCIOTrackerHit();
```
Comment (mirguest) [inline Service/TrackSystemSvc/src/DiagnosticsController.cc.remove:422]: @zhaomr13 This file should be removed.

**Review by mirguest — CHANGES_REQUESTED** [review]
Please make sure the files such as `.remove` are removed.

File: Utilities/KiTrack/src/Tools/VXDHelixFitter.cc.bak [hunk 3]
```diff
@@ -13,7 +13,7 @@
 #include "Tools/KiTrackMarlinTools.h"
 
 
-VXDHelixFitter::VXDHelixFitter( std::vector< TrackerHit* > trackerHits ){
```
Comment (mirguest) [inline Utilities/KiTrack/src/Tools/VXDHelixFitter.cc.bak:16]: Please remove this file.

**Comment by zhaomr13** [comment]
Hi Tao(@mirguest) ,

The files has been removed.

Mingrui.

**Review by mirguest — APPROVED** [review]

**Comment by mirguest** [comment]
Hi @zhaomr13 
Thank you. It's merged now.
Tao

**Pull request merged**